### PR TITLE
Revised 2.0 API

### PR DESCRIPTION
**Pull request by hideki**

### CODE REVIEW

File: shared/src/main/java/com/couchbase/lite/ReadOnlyDictionary.java [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    // TODO: Once Iterable is implemented. This method should be changed to package level access
+    public List<String> allKeys() {
+        if (keys == null) {
+            List<String> results = new ArrayList<>();
+            if (flDict != null) {
+                FLDictIterator itr = new FLDictIterator();
+                itr.begin(flDict);
+                String key;
+                while ((key = SharedKeys.getKey(itr, this.sharedKeys)) != null) {
+                    //while ((key = itr.getKey().asString()) != null) {
```
Comment (pasin) [inline shared/src/main/java/com/couchbase/lite/ReadOnlyDictionary.java]: Remove the comment.

File: shared/src/main/java/com/couchbase/lite/ReadOnlyDocument.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    //---------------------------------------------
+
+    /**
+     * return the document's ID.
+     *
+     * @return the document's ID
+     */
+    public String getId() {
+        if (id == null)
+            id = Misc.CreateUUID();
```
Comment (pasin) [inline shared/src/main/java/com/couchbase/lite/ReadOnlyDocument.java]: ReadOnlyDocument should not generate a new ID. I understand that when creating a new Document() with null ID, the document will create the ID. Please check.

File: shared/src/main/java/com/couchbase/lite/DataUtils.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    public static final float DEFAULT_FLOAT = 0.0f;
+    public static final double DEFAULT_DOUBLE = 0.0d;
+    public static final boolean DEFAULT_BOOLEAN = false;
+
+    public static Date toDate(String string) {
+        Date date = null;
+        return date;
+    }
+
+    public static void validateType(Object value) {
```
Comment (hideki) [inline shared/src/main/java/com/couchbase/lite/DataUtils.java]: I will remove with DB007

File: shared/src/main/java/com/couchbase/lite/CBLData.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            dict.addChangeListener(listener);
+            return dict;
+        } else if (value instanceof List) {
+            Array array = new Array((List) value);
+            array.addChangeListener(listener);
+            return array;
+        } else if (value instanceof Date) {
+            return DateUtils.toJson((Date) value);
+        }
+        return value;
```
Comment (hideki) [inline shared/src/main/java/com/couchbase/lite/CBLData.java]: DB8

File: shared/src/main/java/com/couchbase/lite/ReadOnlyDictionary.java (hunk 1 above)
Comment (hideki) [inline shared/src/main/java/com/couchbase/lite/ReadOnlyDictionary.java]: deleted

File: shared/src/main/java/com/couchbase/lite/ReadOnlyDocument.java (hunk 2 above)
Comment (hideki) [inline shared/src/main/java/com/couchbase/lite/ReadOnlyDocument.java]: fixed in DB7

**Pull request merged**